tests/runner/src/env.rs

```diff
@@ -635,3 +739,117 @@ impl Display for ResultDisplayer {
         }
     }
 }
+
+struct PostgresqlFormatter {
+    pub rows: Vec<PgRow>,
+}
+
+impl Display for PostgresqlFormatter {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.rows.is_empty() {
+            return f.write_fmt(format_args!("(Empty response)"));
+        }
+
+        // create schema
+        let schema = match &self.rows[0] {
+            PgRow::CommandComplete(affected_rows) => {
+                write!(
+                    f,
+                    "{}",
+                    ResultDisplayer {
+                        result: Ok(Output::new_with_affected_rows(*affected_rows as usize)),
+                    }
+                )?;
+                return Ok(());
+            }
+            PgRow::RowDescription(desc) => Arc::new(Schema::new(
+                desc.iter()
+                    .map(|column| {
+                        ColumnSchema::new(column.name(), ConcreteDataType::string_datatype(), false)
+                    })
+                    .collect(),
+            )),
+            _ => unreachable!(),
+        };
+        if schema.num_columns() == 0 {
+            return Ok(());
+        }
+
+        // convert to string vectors
+        let mut columns: Vec<StringVectorBuilder> = (0..schema.num_columns())
+            .map(|_| StringVectorBuilder::with_capacity(schema.num_columns()))
+            .collect();
+        for row in self.rows.iter().skip(1) {
+            if let PgRow::Row(row) = row {
+                for (i, column) in columns.iter_mut().enumerate().take(schema.num_columns()) {
+                    column.push(row.get(i));
+                }
+            }
+        }
+        let columns: Vec<VectorRef> = columns
+            .into_iter()
+            .map(|mut col| Arc::new(col.finish()) as VectorRef)
+            .collect();
+
+        // construct recordbatch
+        let recordbatches = RecordBatches::try_from_columns(schema, columns)
+            .expect("Failed to construct recordbatches from columns. Please check the schema.");
+        let result_displayer = ResultDisplayer {
+            result: Ok(Output::new_with_record_batches(recordbatches)),
+        };
+        write!(f, "{}", result_displayer)?;
+
+        Ok(())
+    }
+}
+
+struct MysqlFormatter {
```

Comment:
Do we have sqlness tests that use the MySQL formatter? We might need to add a test for this.

Reply:
I need some help from @sunng87. He've mentioned there're also some mysql specific behavior https://github.com/GreptimeTeam/greptimedb/pull/4580#issuecomment-2312991800

Reply:
you can use either of 

- `show @@tx_isolation`
- `show @@version_comment`

to test mysql specific interface